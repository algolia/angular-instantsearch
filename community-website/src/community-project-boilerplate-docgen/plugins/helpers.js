const highlight = require('./syntaxHighlighting.js');
const md = require('./mdRenderer.js');

// this plugin provides ATM one helper to easily compute the publicPath of assets
module.exports = function helpers(filenames, metalsmith, cb) {
  // eslint-disable-next-line no-param-reassign
  metalsmith.metadata().h = {
    markdown(src) {
      return md.render(src);
    },
    highlight(src, opts) {
      const lang = opts && opts.lang;
      return highlight(src, lang);
    },
    maybeActive(navPath, singlePathOrArrayOfPaths) {
      const pathsToTest = [].concat(singlePathOrArrayOfPaths);
      return pathsToTest.some(pathToTest => navPath.indexOf(pathToTest) === 0)
        ? 'active'
        : '';
    },
  };

  cb();
};
