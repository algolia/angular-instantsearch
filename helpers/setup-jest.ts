import "jest-preset-angular";

// stupid workaround for TS
import * as helper from "algoliasearch-helper";
require("algoliasearch-helper").default = helper;

const mock = () => {
  let storage = {};
  return {
    getItem(key) {
      return storage[key] || null;
    },
    setItem(key, value) {
      storage[key] = value;
    },
    removeItem(key) {
      delete storage[key];
    },
    clear() {
      storage = {};
    }
  };
};

Object.defineProperty(window, "localStorage", { value: mock() });
Object.defineProperty(window, "sessionStorage", { value: mock() });
Object.defineProperty(window, "getComputedStyle", {
  value: () => ["-webkit-appearance"]
});
