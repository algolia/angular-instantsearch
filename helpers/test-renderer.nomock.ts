import { Component, ViewChild } from '@angular/core';
import { TestBed } from '@angular/core/testing';

import { NgAisInstantSearchModule } from '../src/instantsearch/instantsearch.module';

// same as test-renderer, but without mocking InstantSearch
// In the future, we want to migrate all tests to this

export function createRenderer({
  template,
  TestedWidget,
  defaultState,
  additionalImports,
  additionalDeclarations,
  methods = {},
}: {
  template: string;
  TestedWidget: any;
  additionalImports?: any[];
  additionalDeclarations?: any[];
  defaultState?: {};
  methods?: {};
}) {
  return function(state?: {}, isFirstRender = false) {
    return render(
      {
        template,
        TestedWidget,
        additionalImports,
        additionalDeclarations,
        state: state ? { ...(defaultState || {}), ...state } : undefined,
        methods,
      },
      isFirstRender
    );
  };
}

function render(
  {
    template,
    TestedWidget,
    additionalImports,
    additionalDeclarations,
    state,
    methods,
  }: {
    template: string;
    TestedWidget: any;
    additionalImports?: any[];
    additionalDeclarations?: any[];
    state?: {};
    methods?: {};
  },
  isFirstRender = false
) {
  @Component({
    template: `
      <ais-instantsearch [config]="config">
        ${template}
      </ais-instantsearch>
    `,
  })
  class TestContainer {
    config = {
      searchClient: {
        search(requests) {
          return Promise.resolve({
            results: requests.map(() => ({ hits: [] })),
          });
        },
      },
      indexName: 'instant_search',
    };
    @ViewChild(TestedWidget) testedWidget;
    constructor() {
      Object.keys(methods).forEach(methodName => {
        this[methodName] = methods[methodName];
      });
    }
  }

  TestBed.configureCompiler({
    preserveWhitespaces: false,
  } as any).configureTestingModule({
    declarations: [
      TestContainer,
      TestedWidget,
      ...(additionalDeclarations || []),
    ],
    imports: [NgAisInstantSearchModule.forRoot(), ...(additionalImports || [])],
  });

  const fixture = TestBed.createComponent(TestContainer);
  fixture.detectChanges();

  if (state) {
    fixture.componentInstance.testedWidget.updateState(state, isFirstRender);
    fixture.detectChanges();
  }

  return fixture;
}
