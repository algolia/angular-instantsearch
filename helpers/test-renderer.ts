import { Component, ViewChild } from "@angular/core";
import { TestBed } from "@angular/core/testing";

import { NgAisInstantSearchModule } from "../src/instantsearch/instantsearch.module";

// mock component, dont create a real instantearch instance
jest.mock("../src/base-widget");
jest.mock("../src/instantsearch/instantsearch");

export function createRenderer({
  template,
  TestedWidget,
  defaultState,
  additionalImports,
  additionalDeclarations
}: {
  template: string;
  TestedWidget: any;
  additionalImports?: any[];
  additionalDeclarations?: any[];
  defaultState?: {};
}) {
  return function(state?: {}, firstRender = false) {
    return render(
      {
        template,
        TestedWidget,
        additionalImports,
        additionalDeclarations,
        state: state ? { ...(defaultState || {}), ...state } : undefined
      },
      firstRender
    );
  };
}

function render(
  {
    template,
    TestedWidget,
    additionalImports,
    additionalDeclarations,
    state
  }: {
    template: string;
    TestedWidget: any;
    additionalImports?: any[];
    additionalDeclarations?: any[];
    state?: {};
  },
  firstRender = false
) {
  @Component({
    template: `
      <ng-ais-instantsearch>
        ${template}
      </ng-ais-instantsearch>
    `
  })
  class TestContainer {
    @ViewChild(TestedWidget) testedWidget;
  }

  TestBed.configureTestingModule({
    declarations: [
      TestContainer,
      TestedWidget,
      ...(additionalDeclarations || [])
    ],
    imports: [NgAisInstantSearchModule.forRoot(), ...(additionalImports || [])]
  });

  const fixture = TestBed.createComponent(TestContainer);
  fixture.detectChanges();

  if (state) {
    fixture.componentInstance.testedWidget.updateState(state, firstRender);
    fixture.detectChanges();
  }

  return fixture;
}
