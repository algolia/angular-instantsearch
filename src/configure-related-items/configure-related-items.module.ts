import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { NgAisConfigureRelatedItems } from './configure-related-items';

@NgModule({
  declarations: [NgAisConfigureRelatedItems],
  entryComponents: [NgAisConfigureRelatedItems],
  exports: [NgAisConfigureRelatedItems],
  imports: [CommonModule],
})
export class NgAisConfigureRelatedItemsModule {}
