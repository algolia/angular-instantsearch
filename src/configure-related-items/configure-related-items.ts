import { Component, Input, Inject, forwardRef, Optional } from '@angular/core';

import { EXPERIMENTAL_connectConfigureRelatedItems } from 'instantsearch.js/es/connectors';
import { BaseWidget } from '../base-widget';
import { NgAisInstantSearch } from '../instantsearch/instantsearch';
import { NgAisIndex } from '../index-widget/index-widget';

@Component({
  selector: 'ais-experimental-configure-related-items',
  template: '',
})
export class NgAisConfigureRelatedItems extends BaseWidget {
  @Input() public hit: object;
  @Input() public matchingPatterns: object;
  @Input() public transformSearchParameters: Function;

  constructor(
    @Inject(forwardRef(() => NgAisIndex))
    @Optional()
    public indexParent: NgAisIndex,
    @Inject(forwardRef(() => NgAisInstantSearch))
    public instantSearchParent: NgAisInstantSearch
  ) {
    super('ExperimentalConfigureRelatedItems');
  }

  public ngOnInit() {
    this.createWidget(EXPERIMENTAL_connectConfigureRelatedItems, {
      hit: this.hit,
      matchingPatterns: this.matchingPatterns,
      transformSearchParameters: this.transformSearchParameters,
    });

    super.ngOnInit();
  }
}
