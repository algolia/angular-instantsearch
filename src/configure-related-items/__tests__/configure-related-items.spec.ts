import { createRenderer } from '../../../helpers/test-renderer';
import { NgAisConfigureRelatedItems } from '../configure-related-items';

describe('ConfigureRelatedItems', () => {
  it('renders markup', () => {
    const render = createRenderer({
      TestedWidget: NgAisConfigureRelatedItems,
      template:
        '<ais-experimental-configure-related-items></ais-experimental-configure-related-items>',
    });

    const fixture = render();

    expect(fixture).toMatchSnapshot();
  });

  it('forwards options', () => {
    const createWidget = jest.spyOn(
      NgAisConfigureRelatedItems.prototype,
      'createWidget'
    );
    const hit = {};
    const matchingPatterns = {};
    const transformSearchParameters = jest.fn();

    const render = createRenderer({
      TestedWidget: NgAisConfigureRelatedItems,
      template:
        '<ais-experimental-configure-related-items></ais-experimental-configure-related-items>',
      defaultState: {
        hit,
        matchingPatterns,
        transformSearchParameters,
      },
    });

    render();

    expect(createWidget.mock.calls[0][1]).toEqual({
      hit,
      matchingPatterns,
      transformSearchParameters,
    });
  });
});
