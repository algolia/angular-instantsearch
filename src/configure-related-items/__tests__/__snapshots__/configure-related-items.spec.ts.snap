// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureRelatedItems renders markup 1`] = `
<ng-component
  testedWidget={[Function NgAisConfigureRelatedItems]}
>
  <ais-instantsearch>
    <ais-experimental-configure-related-items />
  </ais-instantsearch>
</ng-component>
`;
